src/main/java/mods/railcraft/common/util/misc/Predicates.java

```diff
@@ -24,7 +25,13 @@
  *
  * @author CovertJaguar <http://www.railcraft.info>
  */
+@SuppressWarnings({"unchecked", "rawtypes"})
 public class Predicates {
+
+    private static final Predicate NON_NULL = Objects::nonNull;
+    private static final Predicate ALWAYS_FALSE = t -> false;
+    private static final Predicate ALWAYS_TRUE = t -> true;
```

Comment:
It does not break. You should use `rawtypes`, just like `java.util.Collections.emptyList()` returns a generic list casted from a rawtype list. [Link to the source] (http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Collections.java#l4458)